mitodl/mitxpro
# #155 Integrate Wagtail Routing

#### Pre-Flight checklist

- [x] Screenshots of screens that were affected

#### What are the relevant tickets?
Closes #155 

#### What's this PR do?
- Adds a migration to move course and program pages to their respective index pages under the home page.
- Moves all properties previously being accessed from Course/Program to CoursePage/ProgramPage
- Moves all page related tests from `courses/models_test` to `cms/models_test`
- Moves templates from `courses/templates` to `cms/templates`
- Modifies partials to use CoursePage/ProgramPage as the source of information
- Updates `serializers` and `serializer_tests` to use the new setup (CoursePage/ProgramPage for source of most information)
- Adds `product_page` template for Wagtail to use in order to serve program and course pages
- Modifies all URL generation code to generate wagtail compatible links for the product pages

#### How should this be manually tested?
Run the migrations, you should end up with a "Courses" page and "Programs" page under the home page, of the types `CourseIndexPage` and `ProgramIndexPage` respectively. All your courses and programs should have been moved to these index pages by the migration. In the CMS, go to any `CoursePage` or `ProgramPage` and click on the `live` link. The page should be rendered as expected, with the url pattern for both being `/courses/<slug>/` and `/programs/<slug>/` respectively.

#### Screenshots (if appropriate)
![Screenshot from 2019-05-29 00-13-22](https://user-images.githubusercontent.com/45350418/58505692-f5ce4100-81a6-11e9-84c9-0ead6ac84a80.png)

![Screenshot from 2019-05-29 00-13-42](https://user-images.githubusercontent.com/45350418/58505696-f9fa5e80-81a6-11e9-966c-47496c876c5b.png)

![Screenshot from 2019-05-29 00-14-13](https://user-images.githubusercontent.com/45350418/58505706-fe267c00-81a6-11e9-9fcc-9beaedb6fca5.png)
